geode-core/src/distributedTest/java/org/apache/geode/internal/cache/control/RebalanceOperationComplexDistributedTest.java

```diff
@@ -0,0 +1,272 @@
+
+import java.io.File;
+import java.io.Serializable;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.apache.commons.io.FileUtils;
+import org.apache.logging.log4j.Logger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.client.ClientCache;
+import org.apache.geode.cache.control.ResourceManager;
+import org.apache.geode.internal.cache.PartitionedRegion;
+import org.apache.geode.logging.internal.log4j.api.LogService;
+import org.apache.geode.test.awaitility.GeodeAwaitility;
+import org.apache.geode.test.dunit.VM;
+import org.apache.geode.test.dunit.rules.ClientVM;
+import org.apache.geode.test.dunit.rules.ClusterStartupRule;
+import org.apache.geode.test.dunit.rules.MemberVM;
+
+/**
+ * The purpose of RebalanceOperationComplexDistributedTest is to test rebalances
+ * across zones and to ensure that enforceUniqueZone behavior of redundancy zones
+ * is working correctly.
+ */
+@RunWith(JUnitParamsRunner.class)
+public class RebalanceOperationComplexDistributedTest implements Serializable {
+  public static final int EXPECTED_BUCKET_COUNT = 113;
+  public static final long TIMEOUT_SECONDS = GeodeAwaitility.getTimeout().getSeconds();
+  public static final String CLIENT_XML = "RebalanceOperationComplex-client.xml";
+  public static final String SERVER_XML = "RebalanceOperationComplex-server.xml";
+  public static final String REGION_NAME = "primary";
+  public static final String COLOCATED_REGION_NAME = "colocated";
+  public static final Logger logger = LogService.getLogger();
+
+  public static final String ZONE_A = "zoneA";
+  public static final String ZONE_B = "zoneB";
+  public int locatorPort;
+  public static final AtomicInteger runID = new AtomicInteger(0);
+  public String workingDir;
+
+  // 6 servers distributed evenly across 2 zones
+  public static final Map<Integer, String> SERVER_ZONE_MAP = new HashMap<Integer, String>() {
+    {
+      put(1, ZONE_A);
+      put(2, ZONE_A);
+      put(3, ZONE_A);
+      put(4, ZONE_B);
+      put(5, ZONE_B);
+      put(6, ZONE_B);
+    }
+  };
+
+  @Rule
+  public ClusterStartupRule clusterStartupRule = new ClusterStartupRule(8);
+
+  @Before
+  public void setup() {
+    // Start the locator
+    MemberVM locatorVM = clusterStartupRule.startLocatorVM(0);
+    locatorPort = locatorVM.getPort();
+
+    workingDir = clusterStartupRule.getWorkingDirRoot().getAbsolutePath();
+
+    runID.incrementAndGet();
+    cleanOutServerDirectories();
+  }
+
+  @After
+  public void after() {
+    stopServersAndDeleteDirectories();
```

Comment:
I think this may be unnecessary, as `ClusterStartupRule` takes care of this clean-up.

Reply:
It doesn't do it right for my test. I have been down that road.

Reply:
The diskdir rule does, but it doesn't interact consistently with the ClusterStartupRule.

Reply:
Oof, that's a pain.